client/history/client.go

```diff
@@ -158,6 +158,7 @@ func (c *clientImpl) DescribeHistoryHost(
 		peer, err = c.peerResolver.FromHostAddress(request.GetHostAddress())
 	}
 	if err != nil {
+		c.logger.Error("peer could not be resolved for host.", tag.Error(err), tag.ShardID(int(request.GetShardIDForHost())), tag.WorkflowID(request.ExecutionForHost.GetWorkflowID()), tag.Address(request.GetHostAddress()))
```

Comment:
`request.ExecutionForHost.GetWorkflowID()` can panic.

may be worth putting this in each branch tbh.

Reply:
on it. tbh I am not sure if there is any other alternative here apart from putting these logs.